### PR TITLE
feat(TDKN-201): Add journal repository for lookup and find operations

### PR DESCRIPTION
**Pull request by fabballe**
**What is the problem this Pull Request is trying to solve?**
 See https://jira.talendforge.org/browse/TDKN-201

**What is the chosen solution to this problem?**
See https://jira.talendforge.org/browse/TDKN-201
 
**Link to the JIRA issue**
https://jira.talendforge.org/browse/TDKN-201
 
**Please check if the Pull Request fulfills these requirements**
- [x] The PR commit message follows our [guidelines](../CONTRIBUTING.md)
- [x] Tests for the changes have been added (for bug fixes / features, coverage should be over 75% in the new code)
- [ ] Docs have been added / updated (for bug fixes / features)
- [x] Related design / discussions / pages (not in JIRA), if any, are all linked or available in the Pull Request

### CODE REVIEW

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/daikon_pull-request_master/39/) [:x:](https://ci-common.datapwn.com/job/daikon_pull-request_master/39/)

File: daikon-content-service/content-service-journal-mongo/src/main/java/org/talend/daikon/content/journal/MongoResourceJournalRepositoryImpl.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+    @Value("${mongo.resource.journal.collection:resourceJournalEntry}")
+    private String collectionName;
+
+    @Autowired
+    private MongoTemplate mongoTemplate;
+
+    @Override
+    public List<ResourceJournalEntry> findByNameStartsWith(String name) {
+        Query findByName = query(where("name").regex(name));
```
Comment (fhuaulme) [inline daikon-content-service/content-service-journal-mongo/src/main/java/org/talend/daikon/content/journal/MongoResourceJournalRepositoryImpl.java]: Shouldn't it be `name + ".*"`for a start with semantic?

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/daikon_pull-request_master/40/) [:x:](https://ci-common.datapwn.com/job/daikon_pull-request_master/40/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/daikon_pull-request_master/41/) [:star:](https://ci-common.datapwn.com/job/daikon_pull-request_master/41/)

File: daikon-content-service/content-service-journal-mongo/src/main/java/org/talend/daikon/content/journal/MongoResourceJournalRepositoryImpl.java (hunk 1 above)
Comment (fabballe) [reply daikon-content-service/content-service-journal-mongo/src/main/java/org/talend/daikon/content/journal/MongoResourceJournalRepositoryImpl.java]: Switching to `"^" + name` syntax

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/daikon_pull-request_master/43/) [:star:](https://ci-common.datapwn.com/job/daikon_pull-request_master/43/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/daikon_pull-request_master/44/) [:star:](https://ci-common.datapwn.com/job/daikon_pull-request_master/44/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/daikon_pull-request_master/45/) [:star:](https://ci-common.datapwn.com/job/daikon_pull-request_master/45/)

**Pull request merged**